S/samtools/build_tarballs.jl

```diff
@@ -0,0 +1,53 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "samtools"
+version = v"1.10.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/samtools/samtools.git", "f01547ee339b68629f754702ba678a63121fbdfa"),
+    DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+for f in ${WORKSPACE}/srcdir/patches/*.patch; do
+    atomic_patch -p1 ${f}
+done
+cd samtools/
+autoheader
+autoconf -Wno-syntax
+      ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make
+make install
+exit
```

Comment:
Judging by the [`README.md`](https://github.com/samtools/samtools/blob/07ad69a6aea7d8b4f40d6a7d73e4c97c5d50f0ef/README.md#building-samtools) this should use something like `--with-htslib=${libdir}` to actually make use of our htslib
```suggestion
./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --with-htslib=${libdir}
make -j${nproc}
make install
```

Reply:
I was wondering about this, and I considered specifying something like `/workspace/x86_64-linux-gnu-libgfortran3-cxx03/destdir/include/htslib/include`, but then the build proceeded without needing to specify so long as I added in the htslib jll file. Do you think we still need to specify `${libdir}`?

Reply:
Setting `CPPFLAGS` was enough to find everything.